packages/ui-lib/src/Misc/molecules/Pagination.tsx

```diff
@@ -0,0 +1,351 @@
+  padding-right: ${WIDTH_PER_NUMBER / 2}px;
+`;
+
+const shakeAnimation = keyframes`
+  0% { transform: translateX(0); }
+  10%, 30%, 50%, 70%, 90% { transform: translateX(-5px); }
+  20%, 40%, 60%, 80% { transform: translateX(5px); }
+  100% { transform: translateX(0); }
+`;
+
+const InputContainer = styled.div<{ borderColor?: string, shouldShake: boolean }>`
+  height: var(--input-height, 40px);
+  animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
+  -moz-animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
+  -webkit-animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
+  -o-animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
+  flex-grow: 0;
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: center;
+  padding: 0 8px;
+  border-radius: 3px;
+  box-shadow: 0 2px 1px 0 rgba(0, 102, 255, 0.04);
+  ${({ borderColor }) => borderColor && `border: 1px solid ${borderColor}`};
+`;
+
+const GoButton = styled(Button)`
+  max-width: 50px;
+  height: 24px;
+  padding: 0 8px;
+  margin-left: 8px;
+`;
+
+const ArrowWrapper = styled.div`
+  display: flex;
+  gap: 0 8px;
+`;
+
+const ArrowButton = styled.button<{ active: boolean }>`
+  ${resetButtonStyles};
+  width: 40px;
+  height: 40px;
+  padding: 12px;
+  border-radius: 3px;
+  box-shadow: 0 4px 9px 0 rgba(152, 174, 189, 0.07);
+  border: solid 1px var(--input-border-default, --grey-9);
+  background-color: var(grey-2);
+  pointer-events: ${({ active }) => active ? 'auto' : 'none'};
+  opacity: ${({ active }) => active ? '1' : '0.5'};
+
+  &:focus, &:hover {
+    outline: 2px solid #838383;
+  }
+`;
+
+const ItemsSelectWrapper = styled.div<{ width: string }>`
+  ${SelectWrapper} {
+    width: ${({ width }) => width ? width : `60px`};
+  }
+    margin-right: 35px;
+`;
+
+export interface IItemsOption {
+  value: number
+  textValue: string
+}
+
+interface OwnProps {
+  pageText?: string
+  totalPages: number
+  activePage?: number
+  buttonText?: string
+  itemsText?: string
+  selectWidth?: string
+  selectDisabled?: boolean
+  itemsOptions: IItemsOption[]
+  onPageChange: (page: number) => void
+  onItemsChange: (items: number) => void
+}
+
+export type IPagination = OwnProps & HTMLAttributes<HTMLDivElement>
+
+const Pagination: React.FC<IPagination> = (props) => {
+  const {
+    pageText = 'Page:',
+    totalPages = 199,
+    activePage = 1,
+    buttonText = 'GO',
+    itemsText = 'Items Per Page',
+    selectWidth = '60px',
+    selectDisabled = false,
+    itemsOptions = [],
+    onPageChange,
+    onItemsChange,
+  } = props;
+
+  const [fieldState, setFieldState] = useState<string>('default');
+  const [pageValue, setPageValue] = useState<string>(activePage.toString());
+  const [disableGO, setDisabledGo] = useState<boolean>(parseInt(pageValue) > totalPages && fieldState !== '' ? false : true);
```

Comment:
This is totally just me being picky, but can be update `disableGO` to `disableGo` for consistency? If you skip this I won't be mad but that case mismatch just jumps out at me 😄